src/containers/register.js

```diff
@@ -1,15 +1,166 @@
+      return Object.assign({}, prevState, {
+        err: false,
+        login: false,
+        register_car: false
+      });
+    });
+    this.props.changePage("/dashboard");
+  }
+
   render() {
     const {} = this.props;
-    const {} = this.state;
-    return <div> </div>;
+    const { login, register_car, car, err } = this.state;
+    return (
+      <div className="register container">
+        <Row>
+          <Col m={6} offset="m3">
+            {login && (
+              <div className="login">
+                <Row>
+                  <h4>Login with Google OAuth</h4>
+                </Row>
+                <Row>
+                  <Col m={6} offset="m3">
+                    <Login login={this.login} />
+                  </Col>
+                </Row>
+              </div>
+            )}
+            {register_car && (
+              <div className="register-car">
+                <h4>Optional: You may register as driver.</h4>
+                <h5>Please enter your car details.</h5>
+                <Row>
+                  <Input
+                    m={12}
+                    label="Make"
+                    value={car.make}
+                    onChange={linkState(this, "car.make")}
+                  />
+                </Row>
+                <Row>
+                  <Input
+                    m={12}
+                    label="Model"
+                    value={car.model}
+                    onChange={linkState(this, "car.model")}
+                  />
+                </Row>
+                <Row>
+                  <Input
+                    m={12}
+                    label="Year"
+                    value={car.year}
+                    onChange={linkState(this, "car.year")}
+                  />
+                </Row>
+                <Row>
+                  <Input
+                    m={12}
+                    label="Color"
+                    value={car.color}
+                    onChange={linkState(this, "car.color")}
+                  />
+                </Row>
+                <Row>
+                  <Col m={2} offset="m2">
+                    <Button
+                      label="Skip"
+                      color="primary"
+                      onClick={() => this.props.changePage("/dashboard")}
+                    />
+                  </Col>
+                  <Col m={2} offset="m4">
+                    <Button
+                      label="Register"
+                      color="secondary"
+                      onClick={this.registerCar}
+                    />
+                  </Col>
+                </Row>
+              </div>
+            )}
+            {err && <span> {err} </span>}
+          </Col>
+        </Row>
+      </div>
+    );
   }
 }
 
-export default RegisterContainer;
+const mapStateToProps = state => {
+  return {
+    ...state.user
+  };
+};
+
+const mapDispatchToProps = dispatch => {
+  return {
+    changePage: async link => await dispatch(push(link)),
```

Comment:
I am pretty sure that this is not needed at all, if react-router-redux was passed into the middleware, then you should be able to just use the Link component and it will automatically update the state for you